tests.py

```diff
@@ -26,66 +35,135 @@ def test_g1_p1(self):
             self.assertAlmostEqual(v, rdata[k])
 
     def test_g2_p2a(self):
-        expected = {'precision': 0.5, 'recall': 1.0, 'f1': 0.6666666666666666, 'tpr': 1.0, 'fpr': 1.0}
+        expected = {'global_precision': 0.5,
+                    'global_recall': 1.0,
+                    'global_f1': 0.6666666666666666,
+                    'global_tpr': 1.0,
+                    'global_fpr': 1.0,
+                    'averaged_precision': 0.4375,
+                    'averaged_recall': 0.875,
```

Comment:
Suggests an edge-case error in our calculations. The `predictions_set2a.tsv` file predicts all items are relevant. So how could the results be anything but `1.0`? Likely because `ground_truth_set2.tsv has:
* One query has zero judgements
* One query has only irrelevant judgements

Probably the latter. I'll take a look at calculations, but perhaps recall is being set to zero if there are zero relevant docs. It should probably be one.

Also though, if a query has zero judged documents, it the query should not be included when counting the average.

Reply:
It is because of the first option. I am counting all queries and one query has all base metrics as `0` hence precision, recall, f1 are also `0`. Therefore, we have an average of 7/8 as `0.875`. I'll exclude queries with zero judged documents from metrics calculation. That would resolve this.

Reply:
Fixed in revision